### DISCUSSION TITLE
weird behavior in step size trace plot during warmup

### DISCUSSION THREAD

**Discussion by ismael-mendoza** [blackjax-devs/blackjax#736] [Q&A] [answered]
I was running multilple the warmup (window_adaptation) of multiple NUTS chains at once using vmap with different random seeds. I observe this behavior with the step sizes during warmup

![image](https://github.com/user-attachments/assets/9a06ba21-534e-4e9d-99c3-7218a5f37089)

And I also see that that at these points there are divergences for all the chains. Is this expected behavior due to the default schedule or is there some weird correlation between my chains? Thank you!

**Comment by junpenglao** [comment]
IIRC this is intended behavior because the step size at the begining of each window will be reset to default.

**Reply by junpenglao** [reply, marked as answer]
To be more precise, the step size will be reset to default at each slow adaptation window: https://github.com/blackjax-devs/blackjax/blob/8a9b5466866cb75b8e2a61f3ac8b7d46b8cc8dca/blackjax/adaptation/window_adaptation.py#L182

**Reply by ismael-mendoza** [reply]
Thanks so much!
